src/database.ts

```diff
@@ -1,10 +1,17 @@
 import { BaseDatabase } from 'adminjs'
 import type { Connection } from 'mongoose'
+import { IAdapterOptions } from './i-adapter-options'
+import { Resource } from './resource'
 
-import Resource from './resource'
+// eslint-disable-next-line no-unused-vars
+type ResourceClass = new (_mongooseModel: any) => Resource;
 
-class Database extends BaseDatabase {
-  private readonly connection: Connection
+abstract class Database extends BaseDatabase {
+  public abstract ResourceClass: ResourceClass;
+
+  public abstract options: IAdapterOptions
+
+  public readonly connection: Connection
```

Comment:
Note had to change this to public as otherwise would get the TS error:
```
error TS4094: Property 'connection' of exported class expression may not be private or protected.
```